### PR TITLE
chore: show suggested widgets in response

### LINKED ISSUE #30968: Move Bindings to a new Bind to Widget button
https://www.notion.so/appsmith/Query-Bind-To-Suggested-Widgets-6d502fed38404c6e8c8752716ee53143?pvs=4

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

Show the Suggested widgets panel in the response pane in the bottom instead of the right side.

This will greatly simplify the Query and Api screens.

- Removes functionality of suggested widgets from the right pane context
- Only renders the right pane if `additionalSections` is passed
- Adds a new Bind Data button in the Response Pane 
- OnClick of button, a menu with suggested widget bindings show up

#### PR fixes following issue(s)
Fixes #30968

#### Media

https://github.com/appsmithorg/appsmith/assets/12022471/20e239ad-dea4-4755-b199-4f0920ae1d46

#### Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Testing

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `BindDataButton` component for easier data binding to widgets in the visual editor.
	- Updated UI elements and interactions in the API and Query editors for improved usability.
	- Added an import statement for `Widgets` from the `DataSources` module.
	- Changed the method call from clicking on a suggested table widget to adding a suggested widget of type `Table`.

- **Refactor**
	- Simplified and streamlined the `ActionRightPane` component by removing outdated functionalities.
	- Adjusted UI components and layouts across various editor components for better consistency and usability.

- **Bug Fixes**
	- Fixed text inconsistency in the `CONNECT_EXISTING_WIDGET_LABEL` constant.
	- Removed unnecessary `useEffect` hook in the `EntityBottomTabs` component to improve performance.

- **Chores**
	- Removed deprecated code and unused imports across several files to clean up the codebase.

### CODE REVIEW

File: app/client/src/ce/constants/messages.ts [hunk 1]
```diff
@@ -740,7 +740,7 @@ export const WELCOME_TOUR_STICKY_BUTTON_TEXT = () => "Next mission";
 export const BINDING_SECTION_LABEL = () => "Bindings";
 export const ADD_NEW_WIDGET_SUB_HEADING = () =>
   "Select how you want to display data.";
-export const CONNECT_EXISTING_WIDGET_LABEL = () => "Select a Widget";
+export const CONNECT_EXISTING_WIDGET_LABEL = () => "Replace on existing widget";
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/ce/constants/messages.ts:743]: @hetunandu is this valid?

**Review by albinAppsmith — DISMISSED** [review]

**Review by albinAppsmith — APPROVED** [review]

**Review by sneha122 — APPROVED** [review]

File: app/client/src/components/editorComponents/ActionRightPane/index.tsx [hunk 2]
```diff
@@ -1,54 +1,14 @@
-import React, { useContext, useEffect, useMemo, useRef } from "react";
```
Comment (AmanAgarwal041) [inline app/client/src/components/editorComponents/ActionRightPane/index.tsx:1]: Are we using ApiRightPane anywhere else ? If not then can we remove it ? cc @hetunandu

Comment (AmanAgarwal041) [reply app/client/src/components/editorComponents/ActionRightPane/index.tsx:1]: Right pane is still used in EE to show inputs of Query Packages

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**